msgpack/fallback.py

```diff
@@ -715,6 +715,20 @@ def _pack(self, obj, nest_limit=DEFAULT_RECURSE_LIMIT,
                 else:
                     raise PackValueError("String is too large")
                 return self._buffer.write(obj)
+            if check(obj, memoryview):
+                n = len(obj) * obj.itemsize
+                if self._use_bin_type:
+                    if n <= 0xff:
+                        self._buffer.write(struct.pack('>BB', 0xc4, n))
+                    elif n <= 0xffff:
+                        self._buffer.write(struct.pack(">BH", 0xc5, n))
+                    elif n <= 0xffffffff:
+                        self._buffer.write(struct.pack(">BI", 0xc6, n))
+                    else:
+                        raise ValueError("memoryview is too large")
+                    return self._buffer.write(obj)
+                else:
+                    raise ValueError("need use_bin_type=True to pack memoryview")
```

Comment:
the problem is that bin type was not there ever and not all code is using use_bin_type=True.

so if one wants to create compatible output with "before bin type" code, that must be possible.

such code does also not use the automatic decoding offered by the Unpacker, because b"foo" and u"foo" data is packed into same output bytes. so, as the unpacker can not know what to decode and what to leave as is, the application needs to know that and do it (or not).